cgen: match true not compiling

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.12 d072437, press to see full `v doctor` output</summary>

|V full version      |V 0.4.12 d29ee738a8efbc1f01a6ae12c432fd869a520b1a.d072437
|:-------------------|:-------------------
|OS                  |windows, Microsoft Windows 11 Pro 22635 64-bit
|Processor           |8 cpus, 64bit, little endian, 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz  
|Memory              |42.51GB/63.67GB
|                    |
|V executable        |D:\software\v\v.exe
|V last modified time|2025-10-20 14:09:59
|                    |
|V home dir          |OK, value: D:\software\v
|VMODULES            |OK, value: C:\Users\Sammy\.vmodules
|VTMP                |OK, value: C:\Users\Sammy\AppData\Local\Temp\v_0
|Current working dir |OK, value: D:\Code\v\unnamed_compiler\v_bugs
|                    |
|Git version         |git version 2.51.1.windows.1
|V git status        |weekly.2025.16-927-gf5a55747
|.git/config present |true
|                    |
|cc version          |cc (Rev2, Built by MSYS2 project) 13.2.0
|gcc version         |gcc (Rev2, Built by MSYS2 project) 13.2.0
|clang version       |clang version 21.1.3
|msvc version        |N/A
|tcc version         |tcc version 0.9.27 (x86_64 Windows)
|tcc git status      |thirdparty-windows-amd64 b425ac82
|emcc version        |emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 4.0.3 (a9651ff57165f5710bb09a5fe52590fd6ddb72df)
|glibc version       |N/A

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  .\match_true_else.v && D:\Code\v\unnamed_compiler\v_bugs\match_true_else.exe`
```v
module main

type SecondTokenizerValue = []rune | Keyword

enum AdvancedTokenType {
	identifier
	keyword
	newline
}

struct SecondTokenizerToken {
	type  AdvancedTokenType
	value ?SecondTokenizerValue
}

enum Keyword {
	module
	import
}

pub struct FileAST {
pub mut:
	module []rune
}

pub fn build_ast(tokens []SecondTokenizerToken) !FileAST {
	mut file_ast := FileAST{}
	mut i := -1

	for i < tokens.len - 1 {
		i++
		token := tokens[i]
		match true {
			file_ast.module.len == 0 && token.type == .keyword
				&& token.value == ?SecondTokenizerValue(Keyword.module) {
				file_ast.module = [`a`]
			}
			file_ast.module.len == 0 {
				return error('Expected `module` keyword at the start of the file, but got `${token.type}`')
			}
			token.type == .keyword && token.value == ?SecondTokenizerValue(Keyword.module) {
				return error('Multiple `module` declarations are not allowed, but got another one')
			}
			token.type == .keyword && token.value == ?SecondTokenizerValue(Keyword.import) {
				dump('import')
			}
			token.type == .newline {
				// ignore newlines
				continue
			}
			else {
				// TODO: turn this into an error after implementing all other AST nodes
				continue
			}
		}
	}

	return file_ast
}

fn main() {
	build_ast([]SecondTokenizerToken{}) or { panic(err) }
}

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: d:/software/v/thirdparty/tcc/include/winapi/synchapi.h:167: warning: CreateMutex redefined
cc: C:/Users/Sammy/AppData/Local/Temp/v_0/match_true_else.01K80ZXM7R1FNYBVGJGRTJE6M3.tmp.c:5603: error: '_t4' undeclared
...
cc: C:/Users/Sammy/AppData/Local/Temp/v_0/match_true_else.01K80ZXM7R1FNYBVGJGRTJE6M3.tmp.c:5603: error: '_t4' undeclared
(note: the original output was 8 lines long; it was truncated to its first 2 lines + the last line)
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

no cgen error


------------

This is a regression starting from f5a55747c0914b6082315fb81985fb3dfbd9b68e
This builds upon https://github.com/vlang/v/pull/25399

(Both were made by @felipensp, so maybe an issue for him)

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/match.v
+++ b/vlib/v/gen/c/match.v
@@ -459,8 +459,6 @@ fn (mut g Gen) match_expr_classic(node ast.MatchExpr, is_expr bool, cond_var str
 					g.write_v_source_line_info(branch)
 					if !reset_if {
 						g.write('else ')
-					} else {
-						reset_if = false
 					}
 				}
 			}

--- a/vlib/v/tests/options/option_match_cases_test.v
+++ b/vlib/v/tests/options/option_match_cases_test.v
@@ -1,0 +1,63 @@
+module main
+
+type SecondTokenizerValue = []rune | Keyword
+
+enum AdvancedTokenType {
+	identifier
+	keyword
+	newline
+}
+
+struct SecondTokenizerToken {
+	type  AdvancedTokenType
+	value ?SecondTokenizerValue
+}
+
+enum Keyword {
+	module
+	import
+}
+
+pub struct FileAST {
+pub mut:
+	module []rune
+}
+
+pub fn build_ast(tokens []SecondTokenizerToken) !FileAST {
+	mut file_ast := FileAST{}
+	mut i := -1
+
+	for i < tokens.len - 1 {
+		i++
+		token := tokens[i]
+		match true {
+			file_ast.module.len == 0 && token.type == .keyword
+				&& token.value == ?SecondTokenizerValue(Keyword.module) {
+				file_ast.module = [`a`]
+			}
+			file_ast.module.len == 0 {
+				return error('Expected `module` keyword at the start of the file, but got `${token.type}`')
+			}
+			token.type == .keyword && token.value == ?SecondTokenizerValue(Keyword.module) {
+				return error('Multiple `module` declarations are not allowed, but got another one')
+			}
+			token.type == .keyword && token.value == ?SecondTokenizerValue(Keyword.import) {
+				dump('import')
+			}
+			token.type == .newline {
+				// ignore newlines
+				continue
+			}
+			else {
+				// TODO: turn this into an error after implementing all other AST nodes
+				continue
+			}
+		}
+	}
+
+	return file_ast
+}
+
+fn test_main() {
+	build_ast([]SecondTokenizerToken{}) or { panic(err) }
+}

